scripts/generate_openapi_document.py

```diff
@@ -0,0 +1,54 @@
+import json
+import os
+from unittest.mock import (
+    PropertyMock,
+    patch,
+)
+
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    config,
+)
+from azul.files import (
+    write_file_atomically,
+)
+from azul.modules import (
+    load_app_module,
+)
+
+
+def main():
+    catalogs = {
+        'dcp2': config.Catalog(name='dcp2',
+                               atlas='hca',
+                               plugins=dict(repository='tdr',
+                                            metadata='hca'))
+    }
+
+    # To create a normalized OpenAPI document, we patch deployment
+    # specific variables that will be present within the document.
+    with patch.object(target=type(config),
+                      attribute='catalogs',
+                      new_callable=PropertyMock,
+                      return_value=catalogs):
+        with patch.object(target=config,
+                          attribute='service_function_name',
+                          return_value='azul_service'):
+            with patch.object(target=config,
+                              attribute='service_endpoint',
+                              return_value='localhost'):
+                app_module = load_app_module('service')
+                assert app_module.config.catalogs == catalogs
+                app_spec = app_module.app.spec()
+                assert app_spec['info']['title'] == 'azul_service'
+                assert one(app_spec['servers'])['url'] == '/'
```

Comment:
The asserts should be made as soon as possible. The first assert could be done in the body of the first `with`, for example.